prow/config/config.go

```diff
@@ -737,6 +752,19 @@ func parseProwConfig(c *Config) error {
 		c.Deck.TideUpdatePeriod = period
 	}
 
+	if c.Deck.Spyglass.SizeLimit == 0 {
+		c.Deck.Spyglass.SizeLimit = 100e6
+	} else if c.Deck.Spyglass.SizeLimit <= 0 {
+		return fmt.Errorf("invalid value for deck.spyglass.size_limit, must be >=0")
+	}
+
+	for k := range c.Deck.Spyglass.Viewers {
+		_, err := regexp.Compile(k)
```

Comment:
we can store the compiled regex in a private field and not re-compile this every request, we do this for e.g. `trigger`

Reply:
I looked through the code in `trigger`, I dont think that exact pattern can be adapted for this case because we have an arbitrary number of regexps. Instead I added a field to the spyglass config struct that caches the compiled regexps, which should have the same effect.